src/components/NavBar.js

```diff
@@ -46,11 +64,18 @@ export default function Navbar() {
         </div>
         <nav className={navbar_menu_container}>
           <ul className={nav_menu}>
-            {navItems.map((buttonText, index) => (
+            {navItems.map((navItem, index) => (
               <li className={nav_item} key={index}>
-                <Link to={buttonText} smooth={true} offset={-navBarHeight} duration={1000}>
-                  <div className={nav_links}>{buttonText}</div>
-                </Link>
+                <div className={nav_item_div}>
+                  <NavLink
+                    to={navItem.urlPath}
+                    className={(isActive) => {
+                      return isActive.isActive ? `${nav_link} ${active}` : nav_link;
```

Comment:
this is wrong.

```typescript
className?: string | ((isActive: boolean) => string) | undefined;
```

here is the definition for what callback is returned. the `isActive` param is just a boolean, and it's not an object.

so, that's why I think it should look like this:

```suggestion
                      return isActive? `${nav_link} ${active}` : nav_link;
```

Reply:
besides, I also dislike how you are returning a concattinated string `${nav_link} ${active}`. why can't it be just `nav_link:active` or something simplier?